metaflow/plugins/metadata/service.py

```diff
@@ -137,8 +127,16 @@ def _start_heartbeat(
         else:
             raise Exception("invalid heartbeat type")
         payload["service_version"] = self.version()
-        msg = Message(MessageTypes.LOG_EVENT, payload)
-        self.sidecar_process.msg_handler(msg)
+        # start sidecar
+        if self.version() is None or LooseVersion(self.version()) < LooseVersion(
+            "2.0.4"
+        ):
+            # if old version of the service is running
+            # then avoid running real heartbeat sidecar process
+            self.sidecar_process = SidecarSubProcess("nullSidecarHeartbeat")
+        else:
+            self.sidecar_process = SidecarSubProcess("heartbeat")
+        self.sidecar_process.send(Message(MessageTypes.CONTEXT, payload))
```

Comment:
Isn't the payload an `Event` here?

Reply:
It doesn't really matter in this case. Happy to make it an EVENT. I considered it to be the start context basically since there are no real events but it doesn't really matter.